src/application-insights/azext_applicationinsights/_client_factory.py

```diff
@@ -16,28 +16,32 @@ def applicationinsights_data_plane_client(cli_ctx, _, subscription=None):
     return ApplicationInsightsDataClient(cred)
 
 
-def applicationinsights_mgmt_plane_client(cli_ctx, _, subscription=None):
+def applicationinsights_mgmt_plane_client(cli_ctx, **kwargs):
     """Initialize Log Analytics mgmt client for use with CLI."""
     from .vendored_sdks.mgmt_applicationinsights import ApplicationInsightsManagementClient
     from azure.cli.core.commands.client_factory import get_mgmt_service_client
-    return get_mgmt_service_client(cli_ctx, ApplicationInsightsManagementClient, subscription_id=subscription)
+    return get_mgmt_service_client(cli_ctx, ApplicationInsightsManagementClient, **kwargs)
 
 
-def cf_query(cli_ctx, _, subscription=None):
-    return applicationinsights_data_plane_client(cli_ctx, _, subscription=subscription).query
+def cf_query(cli_ctx, _):
+    return applicationinsights_data_plane_client(cli_ctx, _).query
 
 
-def cf_metrics(cli_ctx, _, subscription=None):
-    return applicationinsights_data_plane_client(cli_ctx, _, subscription=subscription).metrics
+def cf_metrics(cli_ctx, _):
+    return applicationinsights_data_plane_client(cli_ctx, _).metrics
 
 
-def cf_events(cli_ctx, _, subscription=None):
-    return applicationinsights_data_plane_client(cli_ctx, _, subscription=subscription).events
+def cf_events(cli_ctx, _):
+    return applicationinsights_data_plane_client(cli_ctx, _).events
 
 
-def cf_components(cli_ctx, _, subscription=None):
-    return applicationinsights_mgmt_plane_client(cli_ctx, _, subscription=subscription).components
+def cf_components(cli_ctx, _):
+    return applicationinsights_mgmt_plane_client(cli_ctx).components
 
 
-def cf_api_key(cli_ctx, _, subscription=None):
-    return applicationinsights_mgmt_plane_client(cli_ctx, _, subscription=subscription).api_keys
+def cf_component_billing(cli_ctx, _):
+    return applicationinsights_mgmt_plane_client(cli_ctx).component_current_billing_features
+
+
+def cf_api_key(cli_ctx, _):
```

Comment:
>_ [](start = 24, length = 1)

I'm curious why we need '_' as a method parameter.

Reply:
based on my shallow understanding right now, this is a limitation of azure_cli. `client` function must have at least two positional parameter. At least almost all of commands write like this. I'm not quite familiar with this part right now.